providers/gitlab_test.go

```diff
@@ -108,6 +127,46 @@ func TestGitLabProviderUsername(t *testing.T) {
 	assert.Equal(t, "FooBar", username)
 }
 
+func TestGitlabProviderProject(t *testing.T) {
```

Comment:
I'm taking a look at these units tests for the first time myself since I haven't used Gitlab -- I see alot of redundant test code (historically before you).

Do you have the bandwidth to try to to get these setup using table driven tests?

Reply:
Hi, is something like this https://github.com/factorysh/oauth2-proxy/commit/f2480880919359a32d6a9137ff7987e868e1cf92, the way you want it ? If so, i'll finish and clean up other things like the asserts you mention and then, delete replaced tests.

Reply:
Everything is updated using table tests.

Reply:
This has some samples: https://github.com/oauth2-proxy/oauth2-proxy/blob/master/pkg/apis/sessions/session_state_test.go

Key point is using the `t.Run` method to take the test case name and make a subtest out of it.

We are also trying wherever possible to migrate new tests to using ginkgo more (you can see an example of adding that to legacy test tables here: https://github.com/oauth2-proxy/oauth2-proxy/blob/d7abd56981c72ab71baf5429e5eefc7406c9677f/pkg/apis/sessions/session_state_test.go#L21)

If you went full Ginkgo BDD, here is a sample of a Test Table with that framework we are using for alot of new tests: https://github.com/oauth2-proxy/oauth2-proxy/blob/master/pkg/requests/result_test.go

Reply:
Thanks for reply, never heard of this framework but it was quite fun to use, full migration done in fecd95624afce2d8aad13bf6a9b4a223813aacc3